pkg/controllers/ipaddressallocation/ipaddressallocation_controller.go

```diff
@@ -0,0 +1,217 @@
+}
+
+func deleteFail(r *IPAddressAllocationReconciler, c *context.Context, o *v1alpha1.IPAddressAllocation, e *error) {
+	r.setReadyStatusFalse(c, o, metav1.Now(), e)
+	r.Recorder.Event(o, v1.EventTypeWarning, common.ReasonFailDelete, fmt.Sprintf("%v", *e))
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteFailTotal, MetricResType)
+}
+
+func updateSuccess(r *IPAddressAllocationReconciler, c *context.Context, o *v1alpha1.IPAddressAllocation) {
+	r.setReadyStatusTrue(c, o, metav1.Now())
+	r.Recorder.Event(o, v1.EventTypeNormal, common.ReasonSuccessfulUpdate, "IPAddressAllocation CR has been successfully updated")
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, MetricResType)
+}
+
+func updateFail(r *IPAddressAllocationReconciler, c *context.Context, o *v1alpha1.IPAddressAllocation, e *error) {
+	r.setReadyStatusFalse(c, o, metav1.Now(), e)
+	r.Recorder.Event(o, v1.EventTypeWarning, common.ReasonFailUpdate, fmt.Sprintf("%v", *e))
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, MetricResType)
+}
+
+func (r *IPAddressAllocationReconciler) setReadyStatusFalse(ctx *context.Context, ipaddressallocation *v1alpha1.IPAddressAllocation, transitionTime metav1.Time, err *error) {
+	conditions := []v1alpha1.Condition{
+		{
+			Type:    v1alpha1.Ready,
+			Status:  v1.ConditionFalse,
+			Message: "NSX IPAddressAllocation could not be created/updated/deleted",
+			Reason: fmt.Sprintf(
+				"error occurred while processing the IPAddressAllocation CR. Error: %v",
+				*err,
+			),
+			LastTransitionTime: transitionTime,
+		},
+	}
+	ipaddressallocation.Status.Conditions = conditions
+	e := r.Client.Status().Update(*ctx, ipaddressallocation)
+	if e != nil {
+		log.Error(e, "unable to update IPAddressAllocation status", "ipaddressallocation", ipaddressallocation)
+	}
+}
+
+func (r *IPAddressAllocationReconciler) setReadyStatusTrue(ctx *context.Context, ipaddressallocation *v1alpha1.IPAddressAllocation, transitionTime metav1.Time) {
+	conditions := []v1alpha1.Condition{
+		{
+			Type:               v1alpha1.Ready,
+			Status:             v1.ConditionTrue,
+			Message:            "NSX IPAddressAllocation has been successfully created/updated",
+			Reason:             "",
+			LastTransitionTime: transitionTime,
+		},
+	}
+	ipaddressallocation.Status.Conditions = conditions
+	e := r.Client.Status().Update(*ctx, ipaddressallocation)
+	if e != nil {
+		log.Error(e, "unable to update IPAddressAllocation status", "ipaddressallocation", ipaddressallocation)
+	}
+}
+
+func (r *IPAddressAllocationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	once.Do(func() { go common.GenericGarbageCollector(make(chan bool), servicecommon.GCInterval, r.collectGarbage) })
+	obj := &v1alpha1.IPAddressAllocation{}
+	log.Info("reconciling ipaddressallocation CR", "ipaddressallocation", req.NamespacedName)
```

Comment:
`ipaddressallocation` -> IPAddressAllocation
same for all the following messages

Reply:
ok, all modified.